docs/api_pins.rst

```diff
@@ -11,68 +11,68 @@ are concerned with. However, some users may wish to take advantage of the
++---------+-----------------------------------------------+-------------------------------------------+
+| rpio    | :class:`gpiozero.pins.rpio.RPIOFactory`       | :class:`gpiozero.pins.rpio.RPIOPin`       |
++---------+-----------------------------------------------+-------------------------------------------+
+| pigpio  | :class:`gpiozero.pins.pigpiod.PiGPIOFactory`  | :class:`gpiozero.pins.pigpiod.PiGPIOPin`  |
++---------+-----------------------------------------------+-------------------------------------------+
+| native  | :class:`gpiozero.pins.native.NativeFactory`   | :class:`gpiozero.pins.native.NativePin`   |
++---------+-----------------------------------------------+-------------------------------------------+
+
+If you need to change the default pin factory from within a script, use the
+``Device._set_pin_factory`` class method, passing in the instance of the new
+factory to use. This is only supported at script startup (replacing the factory
+closes all existing pin instances which can have interesting consequences for
+any devices using them)::
+
+    from gpiozero.pins.native import NativeFactory
+    from gpiozero import *
+    Device._set_pin_factory(NativeFactory())
 
     from gpiozero import LED
 
     # This will now use NativePin instead of RPiGPIOPin
     led = LED(16)
 
-``pin_factory`` is a concrete descendent of the abstract :class:`Pin` class.
-The descendent may take additional parameters in its constructor provided they
-are optional; GPIO Zero will expect to be able to construct instances with
-nothing more than an integer pin number.
-
-However, the descendent may take default information from additional sources.
+Certain factories may take default information from additional sources.
 For example, to default to creating pins with
 :class:`gpiozero.pins.pigpiod.PiGPIOPin` on a remote pi called ``remote-pi``
 you can set the :envvar:`PIGPIO_ADDR` environment variable when running your
-script::
-
-    $ PIGPIO_ADDR=remote-pi python my_script.py
-
-It is worth noting that instead of passing an integer to device constructors,
-you can pass an object derived from :class:`Pin` itself::
-
-    from gpiozero.pins.native import NativePin
-    from gpiozero import LED
-
-    led = LED(NativePin(16))
-
-In future, this separation of pins and devices should also permit the library
-to utilize pins that are part of IO extender chips. For example::
-
-    from gpiozero import IOExtender, LED
+script:
 
-    ext = IOExtender()
-    led = LED(ext.pins[0])
-    led.on()
+.. code-block:: console
 
-.. warning::
-
-    While the devices API is now considered stable and won't change in
-    backwards incompatible ways, the pins API is *not* yet considered stable.
-    It is potentially subject to change in future versions. We welcome any
-    comments from testers!
+    $ export GPIOZERO_PIN_FACTORY=pigpio
+    $ PIGPIO_ADDR=remote-pi python3 my_script.py
```

Comment:
I wonder if doing this as a one-liner would be more useful?
`GPIOZERO_PIN_FACTORY=pigpio PIGPIO_ADDR=remote-pi python3 my_script.py`

Less danger of the user getting confused if they forget to un-set GPIOZERO_PIN_FACTORY ;-)